src/sentry/tasks/integrations/slack/find_channel_id_for_rule.py

```diff
@@ -36,13 +39,6 @@ def find_channel_id_for_rule(
         redis_rule_status.set_value("failed")
         return
 
-    user = None
```

Comment:
@markstory 
Previously, this check helped inform the RuleActivity check below.  Using IntegrityError captures the same meaning but without the lookup.  In the future, when this fk is broken for real, it will become a service call and no other enforcement.  This was the intermediate until the FK is broken because technically an integrity error can be raised so long as it remains.